src/main/java/com/cloudant/http/HttpConnection.java

```diff
@@ -0,0 +1,367 @@
+        String string = IOUtils.toString(is);
+        is.close();
+        connection.disconnect();
+        return string;
+    }
+
+    /**
+     * <p>
+     * Return response body data from server as a byte array.
+     * </p>
+     * <p>
+     * <b>Important:</b> you must call <code>execute()</code> before calling this method.
+     * </p>
+     * @return Byte array of response body data from server, if any
+     * @throws IOException if there was a problem reading data from the server
+     */
+    public byte[] responseAsBytes() throws IOException {
+        if (connection == null) {
+            throw new IOException("Attempted to read response from server before calling execute()");
+        }
+        InputStream is = connection.getInputStream();
+        byte[] bytes = IOUtils.toByteArray(is);
+        is.close();
+        connection.disconnect();
+        return bytes;
+    }
+
+    /**
+     * <p>
+     * Return response body data from server as an InputStream.
+     * </p>
+     * <p>
+     * <b>Important:</b> you must call <code>execute()</code> before calling this method.
+     * </p>
+     * @return InputStream of response body data from server, if any
+     * @throws IOException if there was a problem reading data from the server
+     */
+    public InputStream responseAsInputStream() throws IOException {
+        if (connection == null) {
+            throw new IOException("Attempted to read response from server before calling execute()");
+        }
+        InputStream is = connection.getInputStream();
+        return is;
+    }
+
+    /**
+     * Get the underlying HttpURLConnection object, allowing clients to set/get properties not
+     * exposed here.
+     * @return HttpURLConnection the underlying {@link HttpURLConnection} object
+     */
+    public HttpURLConnection getConnection() {
+        return connection;
+    }
+
+    /**
+     * Disconnect the underlying HttpURLConnection. Equivalent to calling:
+     * <code>
+     * getConnection.disconnect()
+     * </code>
+     */
+    public void disconnect() {
+        connection.disconnect();
+    }
+
+    private static String getUserAgent() {
+        String userAgent;
+        String ua = getUserAgentFromResource();
+        {
+            userAgent = String.format("%s Java (%s; %s; %s)",
+                    ua,
+                    System.getProperty("os.arch"),
+                    System.getProperty("os.name"),
+                    System.getProperty("os.version"));
+        }
+        return userAgent;
+    }
+
+    private static String getUserAgentFromResource() {
+        final String defaultUserAgent = "CloudantSync";
+        final URL url = HttpConnection.class.getClassLoader().getResource("mazha.properties");
+        final Properties properties = new Properties();
+        try {
+            //Resources.asByteSource(url) deprecated and removed in Guava 18.0
+            properties.load(Resources.newInputStreamSupplier(url).getInput());
```

Comment:
I suggest using `url.openStream()` for this and avoiding the dependency on guava.

I also think we will need to do some merging with the existing java-cloudant user-agent, but that part can be addressed later.